doc/devel/documenting_mpl.rst

```diff
@@ -492,11 +325,92 @@ For everything but top level chapters, please use ``Upper lower`` for
 section titles, e.g., ``Possible hangups`` rather than ``Possible
 Hangups``
 
+Setters, getters, and keyword arguments
+=======================================
+
+In addition to the "narrative" documentation described above,
+Matplotlib also defines its API reference documentation in docstrings.
+For the most part, these are standard Python docstrings, but
+Matplotlib also includes some features to better support documenting
+getters and setters.
+
+Matplotlib uses artist introspection of docstrings to support properties.
+All properties that you want to support through `~.pyplot.setp` and
+`~.pyplot.getp` should have a ``set_property`` and ``get_property`` method in
+the `~.matplotlib.artist.Artist` class. The setter methods use the docstring
+with the ACCEPTS token to indicate the type of argument the method accepts.
+e.g., in `.Line2D`:
+
+.. code-block:: python
+
+  # in lines.py
+  def set_linestyle(self, linestyle):
+      """
+      Set the linestyle of the line
+
+      ACCEPTS: [ '-' | '--' | '-.' | ':' | 'steps' | 'None' | ' ' | '' ]
+      """
+
+Since Matplotlib uses a lot of pass-through ``kwargs``, e.g., in every function
+that creates a line (`~.pyplot.plot`, `~.pyplot.semilogx`, `~.pyplot.semilogy`,
+etc...), it can be difficult for the new user to know which ``kwargs`` are
+supported.  Matplotlib uses a docstring interpolation scheme to support
+documentation of every function that takes a ``**kwargs``.  The requirements
+are:
+
+1. single point of configuration so changes to the properties don't
+   require multiple docstring edits.
+
+2. as automated as possible so that as properties change, the docs
+   are updated automatically.
+
+The function `matplotlib.artist.kwdoc` and the decorator
+`matplotlib.docstring.dedent_interpd` facilitate this.  They combine Python
+string interpolation in the docstring with the Matplotlib artist introspection
+facility that underlies ``setp`` and ``getp``.  The ``kwdoc`` function gives
+the list of properties as a docstring. In order to use this in another
+docstring, first update the ``matplotlib.docstring.interpd`` object, as seen in
+this example from `matplotlib.lines`:
+
+.. code-block:: python
+
+  # in lines.py
+  docstring.interpd.update(Line2D=artist.kwdoc(Line2D))
+
+Then in any function accepting `~.Line2D` pass-through ``kwargs``, e.g.,
+`matplotlib.axes.Axes.plot`:
+
+.. code-block:: python
+
+  # in axes.py
+  @docstring.dedent_interpd
+  def plot(self, *args, **kwargs):
+      """
+      Some stuff omitted
+
+      The kwargs are Line2D properties:
+      %(Line2D)s
+
+      kwargs scalex and scaley, if defined, are passed on
+      to autoscale_view to determine whether the x and y axes are
+      autoscaled; default True.  See Axes.autoscale_view for more
+      information
+      """
+      pass
```

Comment:
the pass is unneeded (syntax is correct without it and it's not the real impl either)